src/dscanner/analysis/run.d

```diff
@@ -900,88 +727,6 @@ MessageSet analyze(string fileName, const Module m, const StaticAnalysisConfig a
 	return set;
 }
 
-private void resolveAutoFixFromCheck(
-	ref AutoFix autofix,
-	BaseAnalyzer check,
-	const Module m,
-	scope const(Token)[] tokens,
-	const AutoFixFormatting formattingConfig
-)
-{
-	import std.sumtype : match;
-
-	autofix.replacements.match!(
-		(AutoFix.ResolveContext context) {
-			autofix.replacements = check.resolveAutoFix(m, tokens, context, formattingConfig);
-		},
-		(_) {}
-	);
-}
-
-void resolveAutoFixes(ref Message message, string fileName,
-	ref ModuleCache moduleCache,
-	scope const(Token)[] tokens, const Module m,
-	const StaticAnalysisConfig analysisConfig,
-	const AutoFixFormatting overrideFormattingConfig = AutoFixFormatting.invalid)
-{
-	resolveAutoFixes(message.checkName, message.autofixes, fileName, moduleCache,
-		tokens, m, analysisConfig, overrideFormattingConfig);
-}
-
-AutoFix.CodeReplacement[] resolveAutoFix(string messageCheckName, AutoFix.ResolveContext context,
-	string fileName,
-	ref ModuleCache moduleCache,
-	scope const(Token)[] tokens, const Module m,
-	const StaticAnalysisConfig analysisConfig,
-	const AutoFixFormatting overrideFormattingConfig = AutoFixFormatting.invalid)
-{
-	AutoFix temp;
-	temp.replacements = context;
-	resolveAutoFixes(messageCheckName, (&temp)[0 .. 1], fileName, moduleCache,
-		tokens, m, analysisConfig, overrideFormattingConfig);
-	return temp.expectReplacements("resolving didn't work?!");
-}
-
-void resolveAutoFixes(string messageCheckName, AutoFix[] autofixes, string fileName,
-	ref ModuleCache moduleCache,
-	scope const(Token)[] tokens, const Module m,
-	const StaticAnalysisConfig analysisConfig,
-	const AutoFixFormatting overrideFormattingConfig = AutoFixFormatting.invalid)
-{
-	import dsymbol.symbol : DSymbol;
-
-	const(AutoFixFormatting) formattingConfig =
-		overrideFormattingConfig is AutoFixFormatting.invalid
-			? analysisConfig.getAutoFixFormattingConfig()
-			: overrideFormattingConfig;
-
-	scope first = new FirstPass(m, internString(fileName), &moduleCache, null);
-	first.run();
-
-	secondPass(first.rootSymbol, first.moduleScope, moduleCache);
-	auto moduleScope = first.moduleScope;
-	scope(exit) typeid(DSymbol).destroy(first.rootSymbol.acSymbol);
-	scope(exit) typeid(SemanticSymbol).destroy(first.rootSymbol);
-	scope(exit) typeid(Scope).destroy(first.moduleScope);
-
-	GC.disable;
-	scope (exit)
-		GC.enable;
-
-	foreach (BaseAnalyzer check; getAnalyzersForModuleAndConfig(fileName, tokens, m, analysisConfig, moduleScope))
-	{
-		if (check.getName() == messageCheckName)
-		{
-			foreach (ref autofix; autofixes)
-				autofix.resolveAutoFixFromCheck(check, m, tokens, formattingConfig);
-			return;
-		}
-	}
-
-	throw new Exception("Cannot find analyzer " ~ messageCheckName
-		~ " to resolve autofix with.");
-}
-
 void improveAutoFixWhitespace(scope const(char)[] code, AutoFix.CodeReplacement[] replacements)
```

Comment:
Should this not be moved as well?

Reply:
This is not related to autofix logic, this is related to formatting output

Reply:
It is related to formatting output that is given by the autofix logic. Separation of modules should be done based on the relationship between the declarations that are moved, not from an implementation perspective.

Reply:
It seems that this is a helper function for autofix. Why should it be left here if autofix calls it?

Reply:
Understood. Will move it to autofix module